### ISSUE TITLE
Custom scoring function example incompatible with scikit-learn v0.22

### ISSUE THREAD

**Issue by josiahjohnston** [EpistasisLab/tpot#993]
TPOT's custom scoring function breaks with scikit-learn v0.22.

## Context of the issue

sklearn refactored some code in the scorer modules, and tpot has the old module names hardcoded in a heuristic. 
[provide more detailed introduction to the issue itself and why it is relevant]

## Process to reproduce the issue

The example on https://epistasislab.github.io/tpot/using/ generates an error. 
```
from tpot import TPOTClassifier
from sklearn.datasets import load_digits
from sklearn.model_selection import train_test_split
from sklearn.metrics.scorer import make_scorer

digits = load_digits()
X_train, X_test, y_train, y_test = train_test_split(digits.data, digits.target,
                                                    train_size=0.75, test_size=0.25)
# Make a custom metric function
def my_custom_accuracy(y_true, y_pred):
    return float(sum(y_pred == y_true)) / len(y_true)

# Make a custom a scorer from the custom metric function
# Note: greater_is_better=False in make_scorer below would mean that the scoring function should be minimized.
my_custom_scorer = make_scorer(my_custom_accuracy, greater_is_better=True)

tpot = TPOTClassifier(generations=5, population_size=20, verbosity=2,
                      scoring=my_custom_scorer)
tpot.fit(X_train, y_train)
```
> ValueError: Scoring function make_scorer(my_custom_accuracy) looks like it is a metric function rather than a scikit-learn scorer. This scoring type was removed in version 0.11. Please update your custom scoring function.

## Expected result

I expect the example from the documentation to work.

## Current result

The example from the documentation generates an error.

## Possible fix

Use duck typing instead of a heuristic based on module & args_list in `tpot.base:TPOTBase._setup_scoring_function`. Something like:
```
looks_like_old_style_func = False
try:
    scoring(np.array([1,2]), np.array([3,4]))
    looks_like_old_style_func = True
except:
    pass
if looks_like_old_style_func:
    raise ValueError('Scoring function {} looks like it is a metric function....')
```
When I edit the tpot code to use that duck-typing, the issue goes away for me and the example works.

**Comment by weixuanfu** [comment]
Thank you for reporting this issue. But I think this should be fixed in latest version of TPOT (0.11.1). Could you please run another test with scikit-learn 0.22.1?

**Comment by josiahjohnston** [comment]
Good call. I just updated to 0.11.1 and it fixed this issue. I think duck typing would probably be more future-proof than checking sklearn private module names, but this works great for now.

Thanks!!
